### PR TITLE
beginning of scanner stuff

### PR DESCRIPTION
**Pull request by rachelriv**
@alexschneider @whusted @juansc Please look at the framework that I've put together for the scanner when you get the chance. It is just a work in progress right now! I just want to know what you guys think of the structure.

### CODE REVIEW

**Comment by alexschneider** [comment]
Seems good for now - We probably won't need sinon though, since I don't think we really have any async code in our scanner.

**Comment by rachelriv** [comment]
I actually ended up needing sinon to test control flow of the scanner (e.g. ensure that the scanner did not call any methods to extract tokens for empty lines/commented out lines)

I'll add tests to it and get a fully functioning scanner tonight. Then everyone can add in edge cases to the tests as well as ensure that I checked for all tokens that we discussed.

**Comment by rachelriv** [comment]
@alexschneider @whusted @juansc 
Alright team, please take a look at this when you get the chance. I wanted to create a framework that would allow us to easily unit test extractions of different types of tokens for the scanner.

I have yet to implement string literals, which I think we will definitely need. We also should discuss whether or not we want tuple/set/map/list literals. I'm not sure if that should be handled in the scanner or the parser. Please take a look at what I have so far and let me know what you guys think! I tried my best to explain how we want the scanner to behave in the `line_scanner_spec.coffee`. 

Have a great night! :tea: :smile:

File: line_scanner.coffee [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+      # continue to next iteration of while loop
+      # once we've been able to extract a token
+      continue if @extractTwoCharacterTokens()
+      continue if @extractOneCharacterTokens()
+      continue if @extractWords()
+      continue if @extractNumericLiterals()
+
+      # return an error if we were not able to 
+      # extract any tokens from the line
+      return {errors: "invalid token bro"}
```
Comment (rachelriv) [inline line_scanner.coffee:32]: obviously we would have better error handling here...
I vote that we create an error module

File: line_scanner.coffee [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+      return true
+    return false
+
+  addWord: (word) ->
+    if word in tokens.reservedWords
+      @addToken {kind: word}
+    else
+      @addToken {kind: 'ID', lexeme: word}
+
+  extractNumericLiterals: ->
```
Comment (rachelriv) [inline line_scanner.coffee:78]: This method got to be a little longer than I anticipated. I'll clean it up later.
